packages/queued-request-controller/src/QueuedRequestMiddleware.ts

```diff
@@ -92,20 +92,38 @@ export const createQueuedRequestMiddleware = ({
       await messenger.call(
         QueuedRequestControllerActionTypes.enqueueRequest,
         async () => {
-          if (req.method === 'wallet_switchEthereumChain') {
+          if (
+            req.method === 'wallet_switchEthereumChain' ||
+            req.method === 'wallet_addEthereumChain'
+          ) {
             return next();
           }
 
-          const isBuiltIn = isNetworkType(networkClientIdForRequest);
-          const networkConfigurationForRequest = messenger.call(
+          const networkClientConfigurationForRequest = messenger.call(
             'NetworkController:getNetworkClientById',
             networkClientIdForRequest,
           ).configuration;
 
-          const currentProviderConfig = messenger.call(
+          const networkControllerState = messenger.call(
             'NetworkController:getState',
-          ).providerConfig;
+          );
+
+          const isBuiltIn = isNetworkType(networkClientIdForRequest);
+          let networkConfigurationForRequest;
+          if (!isBuiltIn) {
+            networkConfigurationForRequest =
+              networkControllerState.networkConfigurations[
+                networkClientIdForRequest
+              ];
+          } else {
+            // if its a built in
+            // Ideally we should be using only networkConfigurations, and networkClientIds &
+            // networkConfiguration.id should be the same thing.
+            networkConfigurationForRequest =
+              networkClientConfigurationForRequest;
```

Comment:
Will this include a nickname? Or do we use [the `network` value](https://github.com/MetaMask/core/blob/19be7ca19921b20db749cc9742ff2a09c30f72fb/packages/network-controller/src/types.ts#L37) in this case?

Reply:
or `type` I guess it looks like in your extension side PR

Reply:
it doesnt, but theres code in switch ethereum chain handler that looks up a nickname for it in a set of its own constants. This will be fixed when we eventually can always use a NetworkConfiguration instead of switching between NetworkClientConfiguration when the network is builtin

Reply:
Maybe  putting the  network as the nickname here would work too and allow us to simplify some code  in extension / mobile. But yeah I think the better  solution is to just always use NetworkConfiguration